modules/core/src/main/java/org/apache/ignite/internal/processors/cache/verify/checker/tasks/PartitionReconciliationProcessorTask.java

```diff
@@ -201,12 +200,17 @@ public PartitionReconciliationJob(VisorPartitionReconciliationTaskArg arg, Local
             }
 
             try {
+                int parallelism = reconciliationTaskArg.parallelism();
+
+                if (parallelism == 0)
+                    parallelism = Runtime.getRuntime().availableProcessors();
```

Comment:
Possible we should initialize default value into the command handler with others?

Reply:
And the second reason, you call above ```  ignite.compute().broadcastAsync(new ReconciliationSessionId(sesId, arg.parallelism())).get();``` with some value, but here may set other.

Reply:
Fixed in "Fixed test hanging with default parallelism" commit